### ISSUE TITLE
This library requires TEXLIVE distribution installed on the computer to be used?

### ISSUE THREAD

**Issue by iMartinezMateu** [manuels/texlive.js#40]
Hi,

I need to develop a LaTeX web application program in Java that compiles LaTeX documents without the need of making external communication with other process, so the main requirement of this application is not having TEXLIVE (or another distribution) installed on the computer where the program is going to be executed. This library requires TEXLIVE distribution installed on the computer or it can work without any dependency?

Thanks.

**Comment by xylo04** [comment]
Yes, this library will work as a self-contained web application and doesn't require any form of LaTeX to be installed on the user's machine. If you take a look at the demo page http://manuels.github.io/texlive.js/ and pop open the network view of your browser's developer tools, you'll see that TeX Live is downloaded into the user's browser and run there. It sounds like this library should fit your requirements, although it does not have Java bindings, only JavaScript.

**Comment by fzimmermann89** [comment]
Hi,

this javascript library allows to run a javascript version of pdftex in
your browser (or any other javascript interpreter). It needs access to all
latex/font/... files included in your .tex. One way to achieve this is to
install texlive. If you wish you can then clean up the texlive installation
by removing all unnecessary files (as done by the makefile). Another way is
to use the texlive folder in the repo. Currently the library access the
files via emscriptens lazy loading. If you are not using a server for all
the texlive files, you might want to look into the different ways to access
files (or include all neccessary files with emscripten on compile)

Felix


2016-02-08 12:40 GMT+01:00 Iván Martínez Mateu <notifications@github.com>:

**Comment by iMartinezMateu** [comment]
Thanks for your answer.

If I use this library in my application, can I later sell the app in the App Store or the GNU license will be a problem?

**Comment by xylo04** [comment]
I'm not a lawyer, but my impression is that GNU license would be a problem if it were packaged into an installed app.

**Comment by fzimmermann89** [comment]
As far as I am concerned you CAN sell it an the google play store IF you
don't use any DRM. If one person buys your app it must be possible for him
to share it by any means (such as copying the apk to a different device)
and one should be able to change parts your app, recompile it and use it
instead. So you MUST provide complete source code of your app.
You will not be able to sell it on the apple store (you cant disable their
DRM).

**Comment by manuels** [comment]
I would agree with @fzimmermann89 interpretation of the GPL.

**Comment by iMartinezMateu** [comment]
I read in an issue that this library is GPL because of TeXlive. If I replace TeXlive with another non-GPL distribution and I use the pdflatex compiler ported here, can I avoid that my app inherits GPL license? So, I can charge for it without any restriction or obligation.

**Comment by manuels** [comment]
> If I replace TeXlive with another non-GPL distribution and I use the pdflatex compiler ported here, can I avoid that my app inherits GPL license?

No, AFAIK the original pdflatex compiler is also licensed under GPL.

**Comment by iMartinezMateu** [comment]
Mmm... so the only way is choosing XeTeX as it has a MIT License (check https://en.wikipedia.org/wiki/XeTeX ) and trying to use emscripten like you did here to porting it to Javascript. Right? Do you know if XeTeX has SyncTeX support and what are the main differences between LaTeX and XeTeX in terms of package compatibility?

**Comment by fzimmermann89** [comment]
No.
Pdftex (and every other latex to PDF Compiler I know) is GPL.
Am 08.02.2016 9:26 nachm. schrieb "Iván Martínez Mateu" <
notifications@github.com>:

> I read in an issue that this library is GPL because of TeXlive. If I
> replace TeXlive with another non-GPL distribution and I use the pdflatex
> compiler ported here, can I avoid that my app inherits GPL license? So, I
> can charge for it without any restriction or obligation.
>

**Comment by fzimmermann89** [comment]
Again: No.
XeTex doesn't produce pdf files. Either way of converting its output to pdf is GPL or CPL (same restrictions).

I am not willing to discuss this matter any further.
My last advice is: If you want to create a latex app, make it opensource. You propably could earn money because of the laziness of people: Some people will not start searching for a way to get your app for free or chose to pay because they like your work.

Any of this is completely outside the scope of texlive.js..
